sql/core/src/main/scala/org/apache/spark/sql/execution/local/SeqScanNode.scala

```diff
@@ -25,25 +25,30 @@ import org.apache.spark.sql.catalyst.expressions.Attribute
  */
 case class SeqScanNode(output: Seq[Attribute], data: Seq[InternalRow]) extends LeafLocalNode {
 
-  private[this] var iterator: Iterator[InternalRow] = _
-  private[this] var currentRow: InternalRow = _
+  override def execute(): OpenCloseRowIterator = new OpenCloseRowIterator {
 
-  override def open(): Unit = {
-    iterator = data.iterator
-  }
+    private var iter: Iterator[InternalRow] = _
```

Comment:
Why not just assign value here(we can use `val` then)? Your changes to `FilterNode` and `ProjectNode` also assign value to `predicate`/`project` immediately instead of in the `open`.